src/main/java/edu/kis/powp/jobs2d/command/utils/JsonCommandLoader.java

```diff
@@ -0,0 +1,38 @@
+package edu.kis.powp.jobs2d.command.utils;
+
+import com.google.gson.stream.JsonReader;
+import edu.kis.powp.jobs2d.command.ComplexCommand;
+import edu.kis.powp.jobs2d.command.utils.entities.JsonCommandList;
+
+import java.util.Optional;
+import java.util.logging.Logger;
+
+public class JsonCommandLoader implements CommandLoader {
+    private final static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+
+    @Override
+    public Optional<ComplexCommand> loadFromFile(String path) {
+        // read file
+        JsonReader reader = JsonCommandReader.read(path);

```

Comment:
Still, mixed resposibiliters.
The solution should not assume that the command will be read from a file.
You should provide more general API, e.g., loadFromString(String input) or loadFromReader(Reader reader)

Reply:
Interface changed to loadFromReader